File: backend/src/main/java/es/udc/paproject/backend/model/services/PlayerService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Game;
import es.udc.paproject.backend.model.entities.Player;
import es.udc.paproject.backend.model.exceptions.IncorrectDniException;
import es.udc.paproject.backend.model.exceptions.IncorrectEmailException;
import es.udc.paproject.backend.model.exceptions.IncorrectPhoneNumberException;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;

public interface PlayerService {
    
    Player addPlayer(Long teamId, String playerName, String primaryLastName, String secondLastName,
    String position, String trends, String phoneNumber, String email, String dni) throws InstanceNotFoundException, IncorrectDniException, IncorrectEmailException, IncorrectPhoneNumberException;

    void changePlayerToTeam(Long playerId, Long teamId) throws InstanceNotFoundException;

    void clearTotalStatistics(Long playerId) throws InstanceNotFoundException;

    List<Player> findPlayersByUserId(Long userId) throws InstanceNotFoundException;

    Player findPlayerByIdOfTeam(Long playerId, Long teamId) throws InstanceNotFoundException;

    Player findPlayerByDniOfTeam(Long teamId, String dni) throws InstanceNotFoundException, IncorrectDniException;
    //FALTA EN FRONT
    List<Player> findPlayersByTraining(Long trainingId) throws InstanceNotFoundException;
    //FALTA EN FRONT
    List<Player> findPlayersByGame(Long gameId) throws InstanceNotFoundException;

    List<Player> findPlayersByCompletedNameOfTeam(Long teamId, String name, String primaryLastName, String secondLastName) throws InstanceNotFoundException;
    
    List<Player> findAPlayersOfTeam(Long teamId) throws InstanceNotFoundException;

    List<Player> findPlayersByPositionAndTeam(Long teamId, String position) throws InstanceNotFoundException;

    List<Player> findPlayersrWithLesionOfTeam(Long teamId) throws InstanceNotFoundException;

    List<Player> findPlayersWithOneTypeLesion(String typeLesion, Long teamId) throws InstanceNotFoundException;

    void removePlayer(Long teamId, Long playerId) throws InstanceNotFoundException;

    Player updatePlayer(Long teamId, Long playerId, String playerName, String primaryLastName, String secondLastName,
    String position, String trends, String phoneNumber, String email, String dni) throws InstanceNotFoundException, IncorrectDniException, IncorrectEmailException, IncorrectPhoneNumberException;
}

File: frontend/src/backend/playerService.js
import {config, appFetch} from './appFetch';

export const findPlayerByIdOfTeam = (playerId, teamId, onSuccess, onErrors) =>
    appFetch(`/players/${playerId}?teamId=${teamId}`, config('GET'), onSuccess, onErrors);

export const findPlayerById = (playerId, onSuccess, onErrors) =>
    appFetch(`/players/${playerId}`, config('GET'), onSuccess, onErrors);

export const findPlayersByUserId = (onSuccess, onErrors) =>
    appFetch('/players/user', config('GET'), onSuccess, onErrors);

export const findPlayerByDniOfTeam = (teamId, dni, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/dni?dni=${dni}`, config('GET'), onSuccess, onErrors);

export const findPlayersByCompletedNameOfTeam = (teamId, playerName, primaryLastName, secondLastName, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/name?name=${playerName}&primaryLastName=${primaryLastName}&secondLastName=${secondLastName}`, config('GET'), onSuccess, onErrors);

export const findAPlayersOfTeam = (teamId, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/team`, config('GET'), onSuccess, onErrors);

export const findPlayersByTraining = (trainingId, onSuccess, onErrors) =>
    appFetch(`/players/${trainingId}/training`, config('GET'), onSuccess, onErrors);

export const findPlayersByGame = (gameId, onSuccess, onErrors) =>
    appFetch(`/players/${gameId}/game`, config('GET'), onSuccess, onErrors);

export const findPlayersByPositionAndTeam = (teamId, position, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/position?position=${position}`, config('GET'), onSuccess, onErrors);

export const findPlayersrWithLesionOfTeam = (teamId, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/lesion`, config('GET'), onSuccess, onErrors);

export const findPlayersWithOneTypeLesion = (teamId, typeLesion, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/typeLesion?typeLesion=${typeLesion}`, config('GET'), onSuccess, onErrors);

export const addPlayer = (teamId, playerName, primaryLastName, secondLastName, position, trends, phoneNumber, email, dni, onSuccess, onErrors) =>
    appFetch(`/players?teamId=${teamId}&playerName=${playerName}&primaryLastName=${primaryLastName}&secondLastName=${secondLastName}&position=${position}&trends=${trends}&phoneNumber=${phoneNumber}&email=${email}&dni=${dni}`, config('POST'), onSuccess, onErrors);

export const changePlayerToTeam = (teamId, playerId, onSuccess, onErrors) =>
    appFetch(`/players/${teamId}/changePlayerToTeam?playerId=${playerId}`, config('POST'), onSuccess, onErrors);

export const clearTotalStatistics = (playerId, onSuccess, onErrors) =>
    appFetch(`/players/${playerId}/clearTotalStatistics`, config('POST'), onSuccess, onErrors);

export const updatePlayer = (playerId, teamId, playerName, primaryLastName, secondLastName, position, trends, phoneNumber, email, dni, onSuccess, onErrors) =>
    appFetch(`/players/${playerId}?teamId=${teamId}&playerName=${playerName}&primaryLastName=${primaryLastName}&secondLastName=${secondLastName}&position=${position}&trends=${trends}&phoneNumber=${phoneNumber}&email=${email}&dni=${dni}`, config('PUT'), onSuccess, onErrors);

export const removePlayer = (playerId, teamId, onSuccess, onErrors) =>
    appFetch(`/players/${playerId}?teamId=${teamId}`, config('DELETE'), onSuccess, onErrors);